src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBaseKernels.hpp

```diff
@@ -741,13 +741,13 @@ class ElementBasedAssemblyKernel
     arraySlice1d< real64 const, compflow::USD_PHASE - 1 > dPhaseVolFrac_dPres = m_dPhaseVolFrac_dPres[ei];
     arraySlice2d< real64 const, compflow::USD_PHASE_DC - 1 > dPhaseVolFrac_dCompDens = m_dPhaseVolFrac_dCompDens[ei];
 
-    arraySlice1d< real64 const, compflow::USD_PHASE - 1 > phaseDensOld = m_phaseDensOld[ei];
+    arraySlice1d< real64 const, multifluid::USD_PHASE - 2 > phaseDensOld = m_phaseDensOld[ei][0];
```

Comment:
Why do you need this extra (dummy?) dimension? (The `[0]` at the end).
That bothers me here and there in the code.

Reply:
It's because the second dimension contains the quadrature point index. The thing is that our flow kernels assume a single integration point since we use piece-wise constant functions for all our FV relevant quantities. We could loop over quadrature points and avoid hard-coding the value 0 but it would add an extra loop that in these kernels would not make much sense. The problem is that the same constitutive model can be used with a discretization scheme that does need that second dimension.